### DISCUSSION TITLE
Add ignore rules for the Unknow tag rule. (For use with Craft CMS)

### DISCUSSION THREAD

**Discussion by jonathanpardon** [VincentLanglet/Twig-CS-Fixer#303] [Q&A] [answered]
Hi, 
I would like to add this CS fixer to my Craft CMS projects.
Craft uses twig templates, but adds some craft specific tags like redirect, paginate, hook, ..
which result in "File is invalid: Unknown "tagname" tag."

How can I configure to ignore these tags?
Thanks!

**Comment by VincentLanglet** [comment]
Hi,

Which version of Twig CS fixer did you try to install ?
Those checks are supposed to be avoided in the 3.0 version.

**Comment by jonathanpardon** [comment]
Hey Vincent,
I see that Craft CMS locked the version of Twig to 3.8.0
That's probably why I got the 2.11 version of this library.
Is there a workaround for this version?
Thanks for the info.

**Reply by VincentLanglet** [reply]
Is there a reason for craft to not use the latest twig versions ? Which twig version do you use ?

I'll need to check if the constraint can be extended.

**Reply by VincentLanglet** [reply]
> I'll need to check if the constraint can be extended.

A lower constraint is making the test fail, so I would prefer that craft bump the version on his side.
But you may still solve your issue following the doc https://github.com/VincentLanglet/Twig-CS-Fixer/blob/main/docs/configuration.md#token-parsers--more by adding manually the custom TwigExtension coming from craft.

**Comment by jonathanpardon** [comment]
I can only guess that they made a lot of twig extensions that aren't upgraded yet to 3.0.
I have the version that is locked by Craft, so twig 3.8 and twig-cs-fixer 2.11

**Reply by VincentLanglet** [reply]
Kinda weird cause bumping from 3.8 to something bigger only require a minor bump which is BC...

**Reply by VincentLanglet** [reply]
I opened a discussion on craft side https://github.com/craftcms/cms/discussions/15568

**Comment by VincentLanglet** [comment, marked as answer]
Could you try the 3.0.2 version @jonathanpardon ?

**Reply by jonathanpardon** [reply]
Works like a charm, thank you!
